cmake/scripts/common/Macros.cmake

```diff
@@ -333,6 +333,7 @@ endmacro()
 #   dependency added to ${SYSTEM_INCLUDES}, ${DEPLIBS} and ${DEP_DEFINES}
 function(core_require_dep)
   foreach(dep ${ARGN})
+    setup_enable_switch()
```

Comment:
required deps do not have an enable_switch.
If you cannot change relevant code to use FOO_FOUND instead, set ENABLE_FOO in the platform file.